stravalib/client.py

```diff
@@ -1909,23 +2022,23 @@ def _fill_buffer(self):
 
         self._page += 1
 
-    def __iter__(self):
+    def __iter__(self) -> BatchedResultsIterator[T]:
         return self
 
-    def _eof(self):
+    def _eof(self) -> NoReturn:
         """ """
         self.reset()
         raise StopIteration
 
-    def __next__(self):
+    def __next__(self) -> T:
         return self.next()
 
-    def next(self):
-        """ """
+    def next(self) -> T:
         if self.limit and self._counter >= self.limit:
             self._eof()
         if not self._buffer:
             self._fill_buffer()
+        assert isinstance(self._buffer, collections.deque)
```

Comment:
Again, out of curiosity: why did you add this assertion?

Reply:
It tells mypy that here buffer cannot be None. I've actually changed to assert it is not None to make it clearer.